stock_picking_cancel_reason/model/stock_picking.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 ProThai Technology Co.,Ltd. (http://prothaitechnology.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    cancel_reason_id = fields.Many2one(
+        comodel_name="stock.picking.cancel.reason",
+        string="Reason for cancellation",
+        readonly=True,
+        ondelete="restrict",
+    )
+    description = fields.Text(string="Description", readonly=True)
+
+
+class StockPickingCancelReason(models.Model):
+    _name = "stock.picking.cancel.reason"
+    _description = "Stock Picking Cancel Reason"
+
```

Comment:
I think if we have reason, we should also have `active` flag.
As reasons can't be removed, but we may not want to use it in the future.

Reply:
Just a simple "active" field will do.

Reply:
Fixed.